Realm/RLMQueryUtil.mm

```diff
@@ -113,6 +141,68 @@ void set_base_table(const Table*) override {}
     }
 };
 
+
+// Equal and ContainsSubstring are used by QueryBuilder::add_string_constraint as the comparator
+// for performing diacritic-insensitive comparisons.
+
+template <NSUInteger options>
+struct Equal {
+    using CaseSensitive = Equal<options & ~NSCaseInsensitiveSearch>;
```

Comment:
It allows for moving the handling of case-sensitive vs case-insensitive inside the `add_constraint` function rather than repeating it for each operator type. `add_constraint` is invoked with the operator type, and then it uses these type aliases to compute the case-sensitive / case-insensitive variants of that operator type.